x-pack/plugins/fleet/server/services/fleet_server_migration.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { KibanaRequest } from 'src/core/server';
+import {
+  ENROLLMENT_API_KEYS_INDEX,
+  ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE,
+  FleetServerEnrollmentAPIKey,
+} from '../../common';
+import { listEnrollmentApiKeys, getEnrollmentAPIKey } from './api_keys/enrollment_api_key_so';
+import { appContextService } from './app_context';
+
+export async function runFleetServerMigration() {
+  const logger = appContextService.getLogger();
+  logger.info('Starting fleet server migration');
+  await migrateEnrollmentApiKeys();
+  logger.info('Fleet server migration finished');
+}
+
+function getInternalUserSOClient() {
+  const fakeRequest = ({
+    headers: {},
+    getBasePath: () => '',
+    path: '/',
+    route: { settings: {} },
+    url: {
+      href: '/',
+    },
+    raw: {
+      req: {
+        url: '/',
+      },
+    },
+  } as unknown) as KibanaRequest;
+
+  return appContextService.getInternalUserSOClient(fakeRequest);
+}
+
+async function migrateEnrollmentApiKeys() {
```

Comment:
not a blocker but would be good to have tests for this

would agent documents need to be migrated too?

Reply:
Yes agents would need to be migrated too, I am probably going to address that and add tests for that in a following PR as this PR is already complex